4/Client/src/ClusterClientHostedService.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Orleans;
+using Orleans.Configuration;
+using Orleans.Runtime;
+
+namespace Client
+{
+    public class ClusterClientHostedService : IHostedService
+    {
+        private readonly ILogger<ClusterClientHostedService> _logger;
+
+        public IClusterClient Client { get; }
+
+        public ClusterClientHostedService(
+            ILogger<ClusterClientHostedService> logger,
+            ILoggerProvider loggerProvider)
+        {
+            _logger = logger;
+            _logger.LogInformation("creating cluster client...");
+            var advertisedDomain = Environment.GetEnvironmentVariable("SILODOMAIN");
+            IPAddress siloAdvertisedIpAddress = null;
+            siloAdvertisedIpAddress = advertisedDomain != null ? Dns.GetHostAddresses(advertisedDomain).First() : GetLocalIpAddress();
```

Comment:
Nitpicking, but we could probably just collapse lines 29 and 30 into one line removing ` = null;`.

Reply:
`Dns.GetHostAddresses` will probably not work locally (outside docker), so in scenarios where you do provide `SILODOMAIN`, this line will most likely throw.